client/jetpack-cloud/sections/golden-token/golden-token-modal/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { Dialog } from '@automattic/components';
+import { useSelector } from 'react-redux';
+import { isJetpackGoldenTokenPendingActivation as isJetpackGoldenTokenPendingActivationSelector } from 'calypso/state/selectors/is-jetpack-golden-token-pending-activation';
+import { GoldenTokenDialog } from '../golden-token-dialog';
+
+import './style.scss';
+
+// Disabled because Dialog requires an onClose prop, but there is not
+// currently anything to do on close here.
```

Comment:
I think this could be a good opportunity to make `onClose()` optional. Should require just adding optional chaining right before the invocation in this single place:

https://github.com/Automattic/wp-calypso/blob/fb0aab8f67a8738d25d5426343df8599935cdd29/packages/components/src/dialog/index.tsx#L78

The other instances seem to already be accessing `onClose()` with optional chaining, which means they'll support `undefined` values.

Makes sense to be done in its own PR.

Reply:
Good idea. I made such a PR here: https://github.com/Automattic/wp-calypso/pull/75247

I put you as the reviewer but feel free to reassign.

Reply:
Reviewed and approved, thanks 👍